infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerAdapter.java

```diff
@@ -1,51 +1,59 @@
 
-public abstract class CorfuReplicationClusterManagerAdapter {
-    @Getter
-    CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService;
+/**
+ * This is the interface for CorfuReplicationSiteManager.
+ * Implementation of this should have following members:
+ * 1. corfuReplicationDiscoveryService that is needed to notify the cluster configuration change.
+ * 2. localEndpoint that has the local node information.
+ *
+ */
+public interface CorfuReplicationClusterManagerAdapter {
 
-    @Getter
-    TopologyConfigurationMsg topologyConfig;
+    /**
+     *   Set the corfuReplicationDiscoveryServiceMember
+     */
+     void setCorfuReplicationDiscoveryService(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService);
 
-    public void connect(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService) {
-        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
-        start();
-    }
+     /**
+     * Set the localEndpoint
+     */
+    void setLocalEndpoint(String endpoint);
 
-    public synchronized TopologyConfigurationMsg fetchTopology() {
-        topologyConfig = queryTopologyConfig();
-        return topologyConfig;
-    }
+    // This is the currentTopology cached at the adapter.
+    public TopologyConfigurationMsg getTopologyConfig();
 
-    /**
-     * Will be called when the cluster change and a new configuration is sent over
-     *
-     * @param newTopologyConfigMsg
-     */
-    public synchronized void updateTopologyConfig(TopologyConfigurationMsg newTopologyConfigMsg) {
-        if (newTopologyConfigMsg.getTopologyConfigID() > topologyConfig.getTopologyConfigID()) {
-            topologyConfig = newTopologyConfigMsg;
-            corfuReplicationDiscoveryService.updateTopology(topologyConfig);
-        }
-    }
+    // This will talk to the real Site Manager and get the most current topology.
+    TopologyConfigurationMsg queryTopologyConfig();
 
-    public void prepareSiteRoleChange() {
-        corfuReplicationDiscoveryService.prepareClusterRoleChange();
-    }
+    // This is called when get a notification of site config change.
+    void updateTopologyConfig(TopologyConfigurationMsg newSiteConfigMsg);
 
-    public int queryReplicationStatus() {
-        return corfuReplicationDiscoveryService.queryReplicationStatus();
-    }
+    // Start the siteManager service
+    void start();
 
-    //TODO: handle the case that queryTopologyConfig return an exception.
-    public abstract TopologyConfigurationMsg queryTopologyConfig();
+    // Stop the siteManger service
+    void shutdown();
 
-    public abstract void start();
 
-    public abstract void shutdown();
+    /**
+     * While doing a site flip, it is the API used to notify the current log
+     * replication node to prepare a site role type change. It will do some
+     * bookkeeping to calculate the number of log entries to be sent over
+     *
+     */
+    void prepareSiteRoleChange();
```

Comment:
can we rename to -> prepareClusterRoleChange

Reply:
actually if this is only called when an Active is transitioning to a Standby then can we just call it prepareToBecomeStandby() ?